### PR TITLE
modernize lib/file.js and calculate downloaded correctly

### PR DESCRIPTION
**Pull request by jimmywarting**
this fixes the calculation of `.files[i].downloaded`

### CODE REVIEW

File: lib/file.js [hunk 1]
```diff
@@ -1,127 +1,123 @@
...
+  get downloaded () {
     if (!this._torrent.bitfield) return 0
-    var downloaded = 0
-    for (var index = this._startPiece; index <= this._endPiece; ++index) {
+    let downloaded = 0
+    for (let index = this._startPiece; index <= this._endPiece; ++index) {
       if (this._torrent.bitfield.get(index)) {
         // verified data
-        downloaded += (index === this._endPiece) ? this._torrent.lastPieceLength : this._torrent.pieceLength
+        downloaded += index === this._endPiece ? this._torrent.lastPieceLength : this._torrent.pieceLength
```
Comment (KayleePop) [inline lib/file.js]: I think these parentheses make the code more readable.

Comment (DiegoRBaquero) [reply lib/file.js]: I'd rather keep them

**Comment by jimmywarting** [comment]
I also notice `file.download` is calculated wrong.

Using the Sintel torrent as an example...
https://instant.io/#08ada5a7a6183aae1e09d831df6748d566095a10
(this can be tested in the console=

It has 6 srt files and 1 small piece of the mp4 file in the first piece
after the first piece is downloaded you can see that the first files are complete but the `progress` and `downloaded` are incorrect

```js
const file = client.get('08ada5a7a6183aae1e09d831df6748d566095a10').files[0]
console.log(file.done) // true
console.log(file.progress) // 39.58777239709443 <- incorrect b/c of download
console.log(file.downloaded) // 65399
console.log(file.length) // 1652
```

I'm working on a fix for that now also

**Comment by DiegoRBaquero** [comment]
Correct, please separate into two PRs, let's keep this one about modernizing the code.

**Comment by jimmywarting** [comment]
ok, closing this then, new pr: #1478

**Pull request closed without merging**